pages/en/lb4/Creating-components.md

```diff
@@ -307,3 +307,296 @@ class EmailComponent {
   }
 }
 ```
+
+## Creating your own servers 
+
+LoopBack 4 has the concept of a Server, which you can use to create your own
+implementations of REST, SOAP, gRPC, MQTT and more. For an overview, see the
+[Server](#Server.html) section.
+
+### A Basic Example
+The concept of servers in LoopBack doesn't actually require listening on ports;
+you can use this pattern to create workers that process data, write logs, or
+whatever you'd like.
+
+Let's create an example server! Please note that this is not a complete example:
+
+```ts
+import {Server} from '@loopback/core';
+import {processReports} from '../util';
+export class ReportProcessingServer implements Server {
+  @inject('database') db: DefaultCrudRepository<Report, int>;
+  interval: NodeJS.Timer;
+  async start() {
+    // Run the report processor every 2 minutes.
+    let window = Date.now();
+    interval = setInterval(async () => {
+      const results = await db.find({
+        createdDate: {
+          $gt: window,
+        },
+      });
+      if (results && results.length > 0) {
```

Comment:
`results.length > 0` isn't needed. `forEach` won't throw on an empty array.

Reply:
Why call it when I can check if it's got elements?
Also, it could end up being defined, but not a collection with a length property.
Mostly just defensive coding, really.

Reply:
Additional point-of-note: The advanced example is real code that I've tested, but this one was largely a thought experiment. It's definitely not robust (there are plenty of holes in the way it looks for reports to process, for example), and it might not even run at all.

Reply:
Do we want to make this code snippet something users can copy and paste and run? It'd be cool to get them started, especially since this is our basic example of a server.

Reply:
If we want this to be copy-pastable, I'd have to expand it to include all of the same boilerplate snippets contained in the advanced example (`index.ts`, `/src/application.ts`, etc)

Reply:
>  it might not even run at all

Please, let's avoid code that looks like it should run but that does not actually run. If you only want to outline the implementation, then use some sort of a pseudo code.

You can also stick a big fat warning saying the file does not compile, but that would look pretty lame, at least to me. We are writing these code examples to be read by many people, I think the time needed to verify that the code examples actually work is very well worth it!